File: Fuentes/Actividades/Entidades/Areas.vb
Imports System.Data.SqlClient

Public Class Areas

    Private id As Integer
    Private nombre As String
    Private clave As String

    Public Property EId() As Integer
        Get
            Return Me.id
        End Get
        Set(value As Integer)
            Me.id = value
        End Set
    End Property
    Public Property ENombre() As String
        Get
            Return Me.nombre
        End Get
        Set(value As String)
            Me.nombre = value
        End Set
    End Property
    Public Property EClave() As String
        Get
            Return Me.clave
        End Get
        Set(value As String)
            Me.clave = value
        End Set
    End Property

    Public Function ObtenerListado() As List(Of Areas)

        Dim lista As New List(Of Areas)
        Try 
            Dim comando As New SqlCommand()
            comando.Connection = BaseDatos.conexionCatalogo
            comando.CommandText = "SELECT * FROM Areas" 
            BaseDatos.conexionCatalogo.Open()
            Dim dataReader As SqlDataReader
            dataReader = comando.ExecuteReader()
            Dim areas As New Areas
            While (dataReader.Read()) 
                areas = New Areas()
                areas.id = Convert.ToInt32(dataReader("Id"))
                areas.nombre = dataReader("Nombre").ToString()
                areas.clave = dataReader("Clave").ToString()
                lista.Add(areas) 
            End While
            BaseDatos.conexionCatalogo.Close()
            Return lista
        Catch ex As Exception
            Throw ex
        End Try

    End Function

    Public Sub Guardar()

        Try
            Dim comando As New SqlCommand()
            comando.Connection = BaseDatos.conexionCatalogo
            comando.CommandText = "INSERT INTO Areas VALUES (@id, @nombre, @clave)"
            comando.Parameters.AddWithValue("@id", Me.EId)
            comando.Parameters.AddWithValue("@nombre", Me.ENombre)
            comando.Parameters.AddWithValue("@clave", Me.EClave)
            BaseDatos.conexionCatalogo.Open()
            comando.ExecuteNonQuery()
            BaseDatos.conexionCatalogo.Close()
        Catch ex As Exception
            Throw ex
        Finally
            BaseDatos.conexionCatalogo.Close()
        End Try

    End Sub

    Public Sub Editar() 

        Try
            Dim comando As New SqlCommand()
            comando.Connection = BaseDatos.conexionCatalogo
            comando.CommandText = "UPDATE Areas SET Nombre=@nombre, Clave=@clave WHERE Id=@id"
            comando.Parameters.AddWithValue("@id", Me.EId)
            comando.Parameters.AddWithValue("@nombre", Me.ENombre)
            comando.Parameters.AddWithValue("@clave", Me.EClave) 
            BaseDatos.conexionCatalogo.Open()
            comando.ExecuteNonQuery()
            BaseDatos.conexionCatalogo.Close()
        Catch ex As Exception 
            Throw ex
        Finally 
            BaseDatos.conexionCatalogo.Close()
        End Try

    End Sub

    Public Sub Eliminar()

        Try 
            Dim comando As New SqlCommand()
            comando.Connection = BaseDatos.conexionCatalogo
            comando.CommandText = "DELETE FROM Areas WHERE Id=@id"
            comando.Parameters.AddWithValue("@id", Me.id)
            BaseDatos.conexionCatalogo.Open()
            comando.ExecuteNonQuery()
            BaseDatos.conexionCatalogo.Close()
        Catch ex As Exception
            Throw ex
        Finally 
            BaseDatos.conexionCatalogo.Close()
        End Try

    End Sub

    Public Function ValidarPorNumero() As Boolean

        Try
            Dim resultado As Boolean = False
            Dim comando As New SqlCommand()
            comando.Connection = BaseDatos.conexionCatalogo
            comando.CommandText = "SELECT * FROM Areas WHERE Id=@id"
            comando.Parameters.AddWithValue("@id", Me.EId)
            BaseDatos.conexionCatalogo.Open()
            Dim dataReader As SqlDataReader
            dataReader = comando.ExecuteReader()
            If (dataReader.HasRows) Then
                resultado = True
            Else
                resultado = False
            End If
            BaseDatos.conexionCatalogo.Close()
            Return resultado
        Catch ex As Exception
            Throw ex
        Finally
            BaseDatos.conexionInformacion.Close()
        End Try

    End Function


End Class

File: Fuentes/Principal/Entidades/Areas.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Text;

namespace Entidades
{
    public class Areas
    {
        
        private int id;
        private string nombre;
        private string clave;

        public int Id
        {
            get { return this.id; }
            set { this.id = value; }
        }
        public string Nombre
        {
            get { return this.nombre; }
            set { this.nombre = value; }
        }
        public string Clave
        {
            get { return this.clave; }
            set { this.clave = value; }
        }

        public List<Areas> ObtenerListadoPorId()
        {

            List<Areas> lista = new List<Areas>();
            try
            {
                SqlCommand comando = new SqlCommand();
                comando.Connection = BaseDatos.conexionCatalogo;
                comando.CommandText = "SELECT * FROM Areas WHERE Id = @id";
                comando.Parameters.AddWithValue("@id", this.id);
                BaseDatos.conexionCatalogo.Open();
                SqlDataReader dataReader = default(SqlDataReader);
                dataReader = comando.ExecuteReader();
                Areas areas = new Areas();
                while ((dataReader.Read()))
                {
                    areas = new Areas();
                    areas.id = Convert.ToInt32(dataReader["Id"]);
                    areas.nombre = dataReader["Nombre"].ToString();
                    areas.clave = dataReader["Clave"].ToString();
                    lista.Add(areas);
                }
                BaseDatos.conexionCatalogo.Close();
                return lista;
            }
            catch (Exception ex)
            {
                throw ex;
            }
            finally
            {
                BaseDatos.conexionCatalogo.Close();
            }

        }

        public List<Areas> ObtenerListado()
        {

            List<Areas> lista = new List<Areas>();
            try
            {
                SqlCommand comando = new SqlCommand();
                comando.Connection = BaseDatos.conexionCatalogo;
                comando.CommandText = "SELECT * FROM Areas";
                BaseDatos.conexionCatalogo.Open();
                SqlDataReader dataReader = default(SqlDataReader);
                dataReader = comando.ExecuteReader();
                Areas areas = new Areas();
                while ((dataReader.Read()))
                {
                    areas = new Areas();
                    areas.id = Convert.ToInt32(dataReader["Id"]);
                    areas.nombre = dataReader["Nombre"].ToString();
                    areas.clave = dataReader["Clave"].ToString();
                    lista.Add(areas);
                }
                BaseDatos.conexionCatalogo.Close();
                return lista;
            }
            catch (Exception ex)
            {
                throw ex;
            }
            finally
            {
                BaseDatos.conexionCatalogo.Close();
            }

        }

    }
}
